plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -314,7 +318,49 @@ private String getTargetCompatibility(Project project, SourceSet sourceSet) {
   private List<String> getCompilerArgs(Project project, SourceSet sourceSet) {
     JavaCompile javaCompile = getJavaCompileTask(project, sourceSet);
     if (javaCompile != null) {
-      return javaCompile.getOptions().getCompilerArgs();
+      CompileOptions options = javaCompile.getOptions();
+
+      try {
+        DefaultJavaCompileSpec specs = new DefaultJavaCompileSpec();
+        specs.setCompileOptions(options);
+
+        if (GradleVersion.current().compareTo(GradleVersion.version("6.6")) >= 0) {
+          if (javaCompile.getOptions().getRelease().isPresent()) {
+            specs.setRelease(javaCompile.getOptions().getRelease().get());
+          }
+        }
+
+        if (specs.getSourceCompatibility() == null && specs.getRelease() == null) {
+          String sourceCompatibility = javaCompile.getSourceCompatibility();
+          if (sourceCompatibility != null) {
+            specs.setSourceCompatibility(sourceCompatibility);
+          }
+        }
+        if (specs.getTargetCompatibility() == null && specs.getRelease() == null) {
+          String targetCompatibility = javaCompile.getTargetCompatibility();
+          if (targetCompatibility != null) {
+            specs.setTargetCompatibility(targetCompatibility);
+          }
+        }
+
+        JavaCompilerArgumentsBuilder builder = new JavaCompilerArgumentsBuilder(specs)
+                .includeMainOptions(true)
+                .includeClasspath(false)
+                .includeSourceFiles(false)
+                .includeLauncherOptions(false);
+        return builder.build();
+      } catch (Exception e) {
+        // DefaultJavaCompileSpec and JavaCompilerArgumentsBuilder are internal so may not exist.
+        // Fallback to returning just the compiler arguments the build has specified.
+        // This will miss a lot of arguments derived from the CompileOptions e.g. sourceCompatibilty
+        // Arguments must be cast and converted to String because Groovy can use GStringImpl
+        // which then throws IllegalArgumentException when passed back over the tooling connection.
+        List<Object> compilerArgs = new LinkedList<>(options.getCompilerArgs());
+        return compilerArgs
```

Comment:
The return of `getCompilerArgs()` is already a `List<String>`, should be no need to call `toString()` for all the elements

Reply:
Directly `return new LinkedList<>(options.getCompilerArgs());`?